### DISCUSSION TITLE
web Hook for revalidate SSG

### DISCUSSION THREAD

**Discussion by loveyandex** [vercel/next.js#32891] [Help] [answered]
we need to be  possible to trigger nextjs by event passing to it to revalidate ISR......
time based revalidate is not efficient

**Comment by leerob** [comment, marked as answer]
I'm incredibly excited to share that Next.js 12.1 was just released, which includes **on-demand Incremental Static Regeneration**! We know many of you have been looking forward to this for a while, including myself, and I'm so happy it is here. 

- [Learn about Next.js 12.1](https://nextjs.org/blog/next-12-1)
- [Watch a demo of on-demand ISR](https://youtu.be/BGexHR1tuOA)
- [View another demo and give us your feedback](https://on-demand-isr.vercel.app/)
- [Read the documentation](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)
